File: backend/app/models/schemas/qa.py
"""Knowledge-base curation related schemas."""

from typing import Any, Dict, List, Optional

from pydantic import BaseModel, Field


class QARequest(BaseModel):
    """Knowledge-base curation request."""

    query: str = Field(..., description="User question")
    max_iterations: int = Field(default=3, ge=1, le=5, description="Maximum iteration count")
    session_id: Optional[str] = Field(None, description="Session id")
    knowledge_base_id: Optional[int] = Field(None, description="Limit retrieval to one KB")
    category_id: Optional[int] = Field(None, description="Limit retrieval to one category")


class IterationRecord(BaseModel):
    """Single iteration record."""

    round: int
    question: str
    original_question: str
    answer: str
    score: float
    passed: bool
    reason: Optional[str] = None
    suggestion: Optional[str] = None


class QAResponse(BaseModel):
    """Knowledge-base curation response."""

    answer: str = Field(..., description="Answer content")
    confidence_score: float = Field(..., description="Confidence score")
    iteration: int = Field(..., description="Actual iteration count")
    retrieved_docs: List[Dict[str, Any]] = Field(
        default_factory=list,
        description="Retrieved document chunks",
    )
    iteration_history: List[Dict[str, Any]] = Field(
        default_factory=list,
        description="Iteration history",
    )
    document_issues: List[Dict[str, Any]] = Field(
        default_factory=list,
        description="Document issues discovered during curation",
    )
    session_id: Optional[str] = None

File: frontend/lib/api/endpoints/qa.ts
import { apiClient } from "../client";

export interface QARequest {
  query: string;
  max_iterations?: number;
  session_id?: string;
}

export interface QAResponse {
  answer: string;
  confidence_score: number;
  iteration: number;
  retrieved_docs: any[];
  session_id?: string;
}

export const qaApi = {
  invoke: (request: QARequest) =>
    apiClient.post<QAResponse>("/api/v1/qa/invoke", request),

  stream: (request: QARequest) =>
    apiClient.postStream("/api/v1/qa/stream", request),
};
